addons/calendar/static/src/js/mail_activity.js

```diff
@@ -0,0 +1,52 @@
+odoo.define('calendar.Activity', function (require) {
+"use strict";
+
+var Activity = require('mail.Activity');
+var Dialog = require('web.Dialog');
+var core = require('web.core');
+var _t = core._t;
+
+Activity.include({
+    _onEditActivity: function (event) {
+        var self = this;
+        var activity_id = $(event.currentTarget).data('activity-id');
+        _.each(this.activities, function(activity) {
+            if(activity.id === activity_id) {
+                if (activity.activity_meeting_type) {
+                    return self._super(event, {
+                        res_model: 'calendar.event',
+                        res_id: activity.calendar_event_id[0]
+                    });
+                } else {
+                    return self._super(event);
+                }
+            }
+        });
+    },
+    _onUnlinkActivity: function (event) {
+        event.preventDefault();
+        var self = this;
+        var activity_id = $(event.currentTarget).data('activity-id');
+        _.each(this.activities, function(activity) {
+            if(activity.id === activity_id) {
+                if (activity.activity_meeting_type) {
+                    Dialog.confirm(self, _t("The activity is linked to an meeting. Deleting it will remove the meeting as well. Do you want to proceed ?"), {
+                        confirm_callback: function () {
+                            return self._rpc({
+                                model: 'calendar.event',
+                                method: 'unlink',
+                                args: [[activity.calendar_event_id[0]]],
+                            }).then(function () {
+                                self._reload({activity: true});
+                            });
+                        }
```

Comment:
It'd be best to override the method and add a context key to delete the event, to avoid making 2 rpc calls for one action.